### PR TITLE
[Commitment] Move OpeningProof and open to evaluation_proof.rs

### PR DESCRIPTION
**Pull request by xiangxiecrypto**
1. Add module `evaluation_proof`
2. Move OpeningProof and open into `evaluation_proof.rs`
3. Change `kimchi/src/prover.rs` accordingly

### CODE REVIEW

File: poly-commitment/src/evaluation_proof.rs [hunk 1]
```diff
@@ -0,0 +1,306 @@
...
+///     plnms: batch of polynomials to open commitments for with, optionally, max degrees
+///     elm: evaluation point vector to open the commitments at
+///     polyscale: polynomial scaling factor for opening commitments in batch
+///     evalscale: eval scaling factor for opening commitments in batch
+///     oracle_params: parameters for the random oracle argument
+///     RETURN: commitment opening proof
+#[allow(clippy::too_many_arguments)]
+#[allow(clippy::type_complexity)]
+#[allow(clippy::many_single_char_names)]
+impl<G: CommitmentCurve> SRS<G> {
```
Comment (mrmr1993) [inline poly-commitment/src/evaluation_proof.rs:24]: I think the comment and annotations above apply to the function below, and not to this whole `impl` block. Can you move this `impl` line up above them?

**Review by mrmr1993 — COMMENTED** [review]
Very nice, thanks! Left one nit

File: poly-commitment/src/evaluation_proof.rs (hunk 1 above)
Comment (xiangxiecrypto) [reply poly-commitment/src/evaluation_proof.rs]: > I think the comment and annotations above apply to the function below, and not to this whole `impl` block. Can you move this `impl` line up above them?

Yes, of course.

**Review by mrmr1993 — APPROVED** [review]

File: poly-commitment/src/evaluation_proof.rs (hunk 1 above)
Comment (mrmr1993) [reply poly-commitment/src/evaluation_proof.rs]: Thank you!

**Comment by mrmr1993** [comment]
@xiangxiecrypto can you run `cargo fmt`? [CI is failing](https://github.com/o1-labs/proof-systems/runs/5527879758?check_suite_focus=true) with some formatting errors.

**Comment by xiangxiecrypto** [comment]
> @xiangxiecrypto can you run `cargo fmt`? [CI is failing](https://github.com/o1-labs/proof-systems/runs/5527879758?check_suite_focus=true) with some formatting errors.

Yes. Not quite familiar with Rust, sorry for that.

**Comment by mrmr1993** [comment]
That did it. Thanks again @xiangxiecrypto!

**Pull request merged**